dev/tests/api-functional/testsuite/Magento/GraphQl/GiftMessage/Cart/GiftMessageTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\GiftMessage\Cart;
+
+use Exception;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\GraphQl\Quote\GetMaskedQuoteIdByReservedOrderId;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+class GiftMessageTest extends GraphQlAbstract
+{
+    /**
+     * @var GetMaskedQuoteIdByReservedOrderId
+     */
+    private $getMaskedQuoteIdByReservedOrderId;
+
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->getMaskedQuoteIdByReservedOrderId = $objectManager->get(GetMaskedQuoteIdByReservedOrderId::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/GiftMessage/_files/quote_with_message.php
+     * @throws NoSuchEntityException
+     * @throws Exception
+     */
+    public function testGiftMessageForCart()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId->execute('message_order_21');
+        $this->requestCartAndAssertResult($maskedQuoteId);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/guest/create_empty_cart.php
+     * @magentoApiDataFixture Magento/GraphQl/Catalog/_files/simple_product.php
+     * @throws NoSuchEntityException
+     * @throws Exception
+     */
+    public function testGiftMessageForCartWithoutMessage()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId->execute('test_quote');
+        $this->requestCartAndAssertResult($maskedQuoteId);
+    }
+
+    /**
+     * Get Gift Message Assertion
+     *
+     * @param string $quoteId
+     *
+     * @return string
+     * @throws Exception
+     */
+    private function requestCartAndAssertResult(string $quoteId)
+    {
+        $query =  <<<QUERY
+{
+    cart(cart_id: "$quoteId") {
+        gift_message {
+            to
+            from
+            message
+        }
+    }
+}
+QUERY;
+        $response = $this->graphQlQuery($query);
+        self::assertArrayHasKey('gift_message', $response['cart']);
```

Comment:
Can we also assert that the values are correct, and not just existing

Reply:
**Done**